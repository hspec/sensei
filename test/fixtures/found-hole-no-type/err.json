{"version":"1.0","ghcVersion":"ghc-9.10.0","span":{"file":"test/fixtures/found-hole-no-type/Foo.hs","start":{"line":4,"column":8},"end":{"line":4,"column":9}},"severity":"Error","code":88464,"message":["Found hole: _ :: FilePath -> IO String","In a stmt of a 'do' block: r <- _ name\nIn the expression:\n  do r <- _ name\n     return r\nIn an equation for `foo':\n    foo name\n      = do r <- _ name\n           return r","Relevant bindings include\n  name :: FilePath\n    (bound at test/fixtures/found-hole-no-type/Foo.hs:3:5)\n  foo :: FilePath -> IO String\n    (bound at test/fixtures/found-hole-no-type/Foo.hs:3:1)\nValid hole fits include\n  foo (bound at test/fixtures/found-hole-no-type/Foo.hs:3:1)\n  readFile\n    (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10\n     (and originally defined in `ghc-internal-9.1001.0:GHC.Internal.System.IO'))\n  readIO\n    with readIO @String\n    (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10\n     (and originally defined in `ghc-internal-9.1001.0:GHC.Internal.System.IO'))\n  return\n    with return @IO @FilePath\n    (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10\n     (and originally defined in `ghc-internal-9.1001.0:GHC.Internal.Base'))\n  fail\n    with fail @IO @String\n    (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10\n     (and originally defined in `ghc-internal-9.1001.0:GHC.Internal.Control.Monad.Fail'))\n  pure\n    with pure @IO @FilePath\n    (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10\n     (and originally defined in `ghc-internal-9.1001.0:GHC.Internal.Base'))\n  (Some hole fits suppressed; use -fmax-valid-hole-fits=N or -fno-max-valid-hole-fits)"],"hints":[]}
