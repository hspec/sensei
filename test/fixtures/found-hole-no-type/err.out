test/fixtures/found-hole-no-type/Foo.hs:4:8: error: [GHC-88464]
    • Found hole: _ :: FilePath -> IO String
    • In a stmt of a 'do' block: r <- _ name
      In the expression:
        do r <- _ name
           return r
      In an equation for `foo':
          foo name
            = do r <- _ name
                 return r
    • Relevant bindings include
        name :: FilePath
          (bound at test/fixtures/found-hole-no-type/Foo.hs:3:5)
        foo :: FilePath -> IO String
          (bound at test/fixtures/found-hole-no-type/Foo.hs:3:1)
      Valid hole fits include
        foo (bound at test/fixtures/found-hole-no-type/Foo.hs:3:1)
        readFile
          (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10
           (and originally defined in `ghc-internal-9.1202.0:GHC.Internal.System.IO'))
        readIO
          with readIO @String
          (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10
           (and originally defined in `ghc-internal-9.1202.0:GHC.Internal.System.IO'))
        return
          with return @IO @FilePath
          (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10
           (and originally defined in `ghc-internal-9.1202.0:GHC.Internal.Base'))
        fail
          with fail @IO @String
          (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10
           (and originally defined in `ghc-internal-9.1202.0:GHC.Internal.Control.Monad.Fail'))
        pure
          with pure @IO @FilePath
          (imported from `Prelude' at test/fixtures/found-hole-no-type/Foo.hs:1:8-10
           (and originally defined in `ghc-internal-9.1202.0:GHC.Internal.Base'))
        (Some hole fits suppressed; use -fmax-valid-hole-fits=N or -fno-max-valid-hole-fits)

